src/app/agent/agent-dashboard/agent-dashboard.component.ts

```diff
@@ -125,21 +235,66 @@ export class AgentDashboardComponent implements OnInit {
     );
   }
 
-  cancelFlight(booking) {
-    console.log(booking);
+  // Cancel flight
 
-    this.service.put("http://127.0.0.1:8083/agent/booking", booking).subscribe(
-      (res) => {
-        console.log(res);
-      },
-      (error) => {
-        alert(error);
-      }
-    );
+  openCancelBooking() {
+    // Change sidebar classes
+    if (
+      document
+        .getElementById("agent-cancel-flight")
+        .classList.contains("side-link-active")
+    )
+      return;
+    if (
+      document
+        .getElementById("agent-book-flight")
+        .classList.contains("side-link-active")
+    )
+      this.changeBookFlightClass();
+    this.changeCancelFlightClass();
   }
 
+  cancelFlight(booking) {
+    this.service
+      .put(`${environment.agentBackendUrl}${environment.bookingUri}`, booking)
+      .subscribe(
+        (res) => {
+          console.log(res);
+        },
+        (error) => {
+          alert(error);
+        }
+      );
+  }
+
+  // Logout
+
   logout() {
     this.authService.logout();
     this.router.navigate(["/agent/login"]);
   }
+
+  // Aesthetics
+  changeCancelFlightClass() {
+    document
+      .getElementById("agent-cancel-flight")
+      .classList.toggle("side-link-active");
+    if (this.cancelFlightPage === true) {
+      this.cancelFlightPage = false;
+    } else if (this.cancelFlightPage === false) {
+      this.cancelFlightPage = true;
+      this.bookFlightPage = false;
+    }
+  }
+
+  changeBookFlightClass() {
+    document
+      .getElementById("agent-book-flight")
+      .classList.toggle("side-link-active");
+    if (this.bookFlightPage === true) {
+      this.bookFlightPage = true;
```

Comment:
This looks like it should be false instead of true.

Reply:
it actually should do nothing but good catch!